mne/io/base.py

```diff
@@ -467,6 +467,7 @@ def _read_segment(self, start=0, stop=None, sel=None, data_buffer=None,
 
         #  Initialize the data and calibration vector
         n_sel_channels = self.info['nchan'] if sel is None else len(sel)
+        assert n_sel_channels <= self.info['nchan']
```

Comment:
do you want to keep this assert?

Reply:
yes it's like an "active comment" (plus prevents us from doing something dumb in a calling function when developing)

Reply:
Don't get me wrong, I'm totally in favor of that kind of asserts.